### PR TITLE
Devise readiness: Simplify login spec helper

### PR DESCRIPTION
**Pull request by johnnyshields**
This PR simplifies the spec helper login macros of Authlogic to match Devise, and makes a few spec changes to reduce changes needed on devise branch.

In this PR I have deleted two specs which are re-added by #671. This is intentional and the code from #671 should be used to resolve the conflict.

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -49,14 +49,14 @@
 
       it "should render the requested user as JSON" do
         expect(User).to receive(:find).and_return(current_user)
-        expect(current_user).to receive(:to_json).and_return("generated JSON")
+        expect_any_instance_of(User).to receive(:to_json).and_return("generated JSON")
```
Comment (CloCkWeRX) [inline spec/controllers/users_controller_spec.rb]: Presumably we don't need 
 expect(User).to receive(:find).and_return(current_user)
if we are no longer stubbing just that record?

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -70,14 +70,14 @@
 
       it "should render the requested user as XML" do
         expect(User).to receive(:find).and_return(current_user)
-        expect(current_user).to receive(:to_xml).and_return("generated XML")
+        expect_any_instance_of(User).to receive(:to_xml).and_return("generated XML")
```
Comment (CloCkWeRX) [inline spec/controllers/users_controller_spec.rb]: Ditto

File: spec/models/entities/opportunity_spec.rb [hunk 3]
```diff
@@ -30,7 +30,7 @@
 require File.expand_path(File.dirname(__FILE__) + '/../../spec_helper')
 
 describe Opportunity do
-  before { login }
+  let(:current_user) { FactoryGirl.create(:user) }
```
Comment (CloCkWeRX) [inline spec/models/entities/opportunity_spec.rb]: Its weird that these tests want a user
```
  it "should be possible to create opportunity with the same name" do
    FactoryGirl.create(:opportunity, name: "Hello", user: current_user)
    expect { FactoryGirl.create(:opportunity, name: "Hello", user: current_user) }.to_not raise_error
  end
```

We should shift that into a context to explain it ("with a logged in user"); put it in the test case name or drop the user requirements

File: spec/controllers/users_controller_spec.rb (hunk 1 above)
Comment (johnnyshields) [reply spec/controllers/users_controller_spec.rb]: I'll address this in the final Devise PR.

**Comment by johnnyshields** [comment]
@CloCkWeRX about the "current_user" in the model specs, the user field (belongs_to user) must be set on most objects. However, this will be handled in the factory and can be removed from the spec. I've raised https://github.com/fatfreecrm/fat_free_crm/issues/677 to track this.

**Pull request merged**